dandi/metadata.py

```diff
@@ -254,6 +254,16 @@ def extract_wasAttributedTo(metadata):
     wat = extract_model(
         models.Participant, metadata, identifier=metadata.get("subject_id")
     )
+    if all(v is None for k, v in wat.dict().items() if k != "schemaKey"):
+        return []
+    else:
+        return [wat]
+
+
+def extract_wasGeneratedBy(metadata):
+    wat = extract_model(
+        models.Session, metadata, identifier=None, name=metadata.get("session_id")
+    )
```

Comment:
@jwodder would you please RF this code to avoid duplication?  it seems we need some helper like `extract_simple(field, metadata)` which could be used as a `partial` instead of 3 identical copies now

Reply:
I factored out the conversion to a list, but the different treatment of the fields in `extract_wasGeneratedBy()` prevents factoring down to a `extra_model_list(modelcls, identifier_field, metadata)` function.

Reply:
but would work as `extract_model_list(modelcls, identifier_field, metadata, **kw)` where you would pass e.g. `identifier=None` through via `kw` (and as a part of the `partial`) right?

Reply:
Code compressed further.